docker/pypi/wmagent/run.sh

```diff
@@ -0,0 +1,684 @@
+        fi
+        [[ $errVal -eq 0 ]] || { echo "ERROR: Failed to populate WMAgent's resource control!"; return $(false) ;}
+    }
+    echo "Done: $stepMsg"
+    echo "-------------------------------------------------------"
+}
+
+agent_upload_config(){
+    # A function to be used for uploading WMAgentConfig to AuxDB at Central CouchDB
+    # NOTE: The final config/.dockerInit is to be set here after full agent initialisation.
+    #       Once we have reached to the step of uploading the agent config to Cerntral CouchDB
+    #       we consider all previous reintialisation and config steps as successfully complpeted.
+    #       Steps to follow:
+    #       * Checking if the current image WMA_BUILD_ID matches the last one successfully initialised at the host
+    #       * If not, tries to upload the current config to Central CouchDB
+    #       * If agent config successfully uploaded, preserve the WMA_BUILD_ID at config/.dockerInit
+    #       With that, we consider the agent initialisation fully complete. The init steps will not be
+    #       repeated on further container restarts unless any of the */.dockerInit files at the host is altered.
+    local stepMsg="Performing $FUNCNAME"
+    echo "-------------------------------------------------------"
+    echo "Start: $stepMsg"
+
+    _init_valid $WMA_CONFIG_DIR || {
+        echo "$FUNCNAME: triggered."
+        echo "$FUNCNAME: Tweaking central agent configuration befre uploading"
+        if [[ "$TEAMNAME" == production ]]; then
+            echo "$FUNCNAME: Agent connected to the production team, setting it to drain mode"
+            agentExtraConfig='{"UserDrainMode":true}'
+        elif [[ "$TEAMNAME" == *testbed* ]]; then
+            echo "$FUNCNAME: Testbed agent, setting MaxRetries to 0..."
+            agentExtraConfig='{"MaxRetries":0}'
+        elif [[ "$TEAMNAME" == *dev* ]]; then
+            echo "$FUNCNAME: Dev agent, setting MaxRetries to 0..."
+            agentExtraConfig='{"MaxRetries":0}'
+        fi
+        ### Upload WMAgentConfig to AuxDB
+        echo "*** Upload WMAgentConfig to AuxDB ***"
+        $manage execute-agent wmagent-upload-config $agentExtraConfig && echo $WMA_BUILD_ID > $WMA_CONFIG_DIR/.dockerInit
+    }
+    echo "Done: $stepMsg"
+    echo "-------------------------------------------------------"
+}
+
+start_services() {
+    local stepMsg="Performing $FUNCNAME"
+    echo "-------------------------------------------------------"
+    echo "Start: $stepMsg"
+    $manage start-services
+    echo "Done: $stepMsg"
+    echo "-------------------------------------------------------"
+}
+
+start_agent() {
+    local stepMsg="Performing $FUNCNAME"
+    echo "-------------------------------------------------------"
+    echo "Start: $stepMsg"
+    echo "-------------------------------------------------------"
+    echo "Start sleeping now ...zzz..."
+    while true; do sleep 10; done
```

Comment:
Why do we have this infinity loop here?

Reply:
Because at this stage (while still the working on the whole containerization in general is ongoing) we'd better connect to the container and run the agent manually with 
```
manage start-agent
```  
It is to be removed once we are ready. I'll make a TODO comment, so we won't forget it.